x-pack/plugins/siem/public/components/open_timeline/index.tsx

```diff
@@ -0,0 +1,233 @@
+        ? [...validSelections, timelineResult.savedObjectId]
+        : validSelections,
+    []
+  );
+
+/** Manages the state (e.g table selection) of the (pure) `OpenTimeline` component */
+export class StatefulOpenTimeline extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      itemIdToExpandedNotesRowMap: {},
+      onlyFavorites: false,
+      search: '',
+      pageIndex: 0,
+      pageSize: props.defaultPageSize,
+      sortField: DEFAULT_SORT_FIELD,
+      sortDirection: DEFAULT_SORT_DIRECTION,
+      selectedItems: [],
+    };
+  }
+
+  public componentDidMount() {
+    this.focusInput();
+  }
+
+  public render() {
+    const {
+      addTimelinesToFavorites,
+      deleteTimelines,
+      defaultPageSize,
+      openTimeline,
+      searchResults,
+      title,
+    } = this.props;
+    const {
+      itemIdToExpandedNotesRowMap,
+      onlyFavorites,
+      pageIndex,
+      pageSize,
+      search: query,
+      selectedItems,
+      sortDirection,
+      sortField,
+    } = this.state;
+
+    // TODO: wrap `OpenTimeline` below with the GraphQL query, and pass
+    // `isLoading`, `searchResults`, `totalSearchResultsCount`, etc:
+    return (
+      <OpenTimeline
+        deleteTimelines={deleteTimelines}
+        defaultPageSize={defaultPageSize}
+        isLoading={false}
+        itemIdToExpandedNotesRowMap={itemIdToExpandedNotesRowMap}
+        onAddTimelinesToFavorites={
+          addTimelinesToFavorites != null ? this.onAddTimelinesToFavorites : undefined
+        }
+        onDeleteSelected={deleteTimelines != null ? this.onDeleteSelected : undefined}
+        onlyFavorites={onlyFavorites}
+        onOpenTimeline={openTimeline}
+        onQueryChange={this.onQueryChange}
+        onSelectionChange={this.onSelectionChange}
+        onTableChange={this.onTableChange}
+        onToggleOnlyFavorites={this.onToggleOnlyFavorites}
+        onToggleShowNotes={this.onToggleShowNotes}
+        pageIndex={pageIndex}
+        pageSize={pageSize}
+        query={query}
+        searchResults={searchResults.slice(pageIndex * pageSize, pageIndex * pageSize + pageSize)}
+        selectedItems={selectedItems}
+        sortDirection={sortDirection}
+        sortField={sortField}
+        title={title}
+        totalSearchResultsCount={searchResults.length}
+      />
+    );
+  }
+
+  /** Invoked when the user presses enters to submit the text in the search input */
+  private onQueryChange: OnQueryChange = (query: EuiSearchBarQuery) => {
+    this.setState({
+      search: query.queryText.trim(),
+    });
+  };
+
+  /** Focuses the input that filters the field browser */
+  private focusInput = () => {
+    const elements = document.querySelector(`.${OPEN_TIMELINE_CLASS_NAME} input`);
+
+    if (elements != null) {
+      (elements as HTMLElement).focus(); // this cast is required because focus() does not exist the base `Element` type returned by `document.querySelector`
```

Comment:
I believe you can pass the type in to `querySelector` instead of this as cast, ala:

``` ts
const elements = document.querySelector<HTMLElement>(`.${OPEN_TIMELINE_CLASS_NAME} input`);
```